### PR TITLE
Add option to enable UOS patch or not

### PR DESCRIPTION
**Pull request by pacoyang**
Solve:

- #193 

Add new option variable to enable UOS patch or not, by default false.

Related to:

- #192
- #195

### CODE REVIEW

File: src/env-vars.ts [hunk 1]
```diff
@@ -38,9 +38,17 @@ const WECHATY_PUPPET_WECHAT_TOKEN = (value?: string): undefined | string => {
   return process.env['WECHATY_PUPPET_WECHAT_TOKEN']
 }
 
+const WECHATY_PUPPET_WECHAT_UOS = (value?: boolean): boolean => {
+  if (typeof value !== 'undefined') {
+    return value
+  }
+  return String(process.env['WECHATY_PUPPET_WECHAT_UOS']) === '1'
```
Comment (huan) [inline src/env-vars.ts]: >   return !!process.env['WECHATY_PUPPET_WECHAT_PUPPETEER_STEALTHLESS']

Like the `STEALTHLESS`, please follow the naming convention and the `!!` boolean check as the following demonstrated:

```diff
const WECHATY_PUPPET_WECHAT_UOS = (value?: boolean): boolean => {
  if (typeof value !== 'undefined') {
    return value
  }
-   return String(process.env['WECHATY_PUPPET_WECHAT_UOS']) === '1'
+   return !!process.env['WECHATY_PUPPET_WECHAT_UOS']
```

File: src/bridge.ts [hunk 2]
```diff
@@ -66,7 +66,8 @@ export interface BridgeOptions {
   launchOptions?  : puppeteer.LaunchOptions,
   memory          : MemoryCard,
   stealthless?    : boolean,
-  extspam?        : string
+  extspam?        : string,
```
Comment (huan) [inline src/bridge.ts:69]: rename it from `extspam` to `uosExtSpam`

File: README.md [hunk 3]
```diff
@@ -85,6 +85,7 @@ We use [stealth](https://www.npmjs.com/package/puppeteer-extra-plugin-stealth) t
 | head          | boolean |     false     | puppeteerlaunchoptions.headless                                                                                             |
 | launchOptions | object  |       -       | same to [puppeteerlaunchoptions](https://github.com/GoogleChrome/puppeteer/blob/v1.18.1/docs/api.md#puppeteerlaunchoptions) |
 | stealthless   | boolean |     false     | disabled [puppeteer-extra-plugin-stealth](https://www.npmjs.com/package/puppeteer-extra-plugin-stealth) or not              |
+| uos           | boolean |     false     | enable [UOS patch](https://github.com/wechaty/puppet-wechat/issues/127) or not                                              |
```
Comment (huan) [inline README.md:88]: Please add `uosExtSpam` to the docs as well.

**Review by huan — CHANGES_REQUESTED** [review]
Thank you very much for adding the UOS switch for enabling/disabling the related codes!

Please follow my review and then it will be ready to be merged.

File: src/env-vars.ts (hunk 1 above)
Comment (pacoyang) [reply src/env-vars.ts]: I change this code to:
```ts
return /^(true|1)$/i.test(String(process.env['WECHATY_PUPPET_WECHAT_PUPPETEER_UOS']))
```
[RegExp.prototype.test](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) returns boolean type.
It is more flexible for user to enable or disable UOS patch through env variable `WECHATY_PUPPET_WECHAT_PUPPETEER_UOS`, like setting `WECHATY_PUPPET_WECHAT_PUPPETEER_UOS=0` to disable.

**Review by huan — APPROVED** [review]
LGTM

**Comment by huan** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited Paco Yang to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message. (See also: [wechaty/PMC#16](https://github.com/wechaty/PMC/issues/16))

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Pull request merged**